javascript/commons/Countdown.js

```diff
@@ -96,62 +96,43 @@ liquipedia.countdown = {
-		}
-		if ( timerObjectNode.dataset.streamHuomao ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/huomao/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamHuomao ) + '"><i class="lp-icon lp-icon-21 lp-huomaotv"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamHuya ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/huya/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamHuya ) + '"><i class="lp-icon lp-icon-21 lp-huyatv"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamKick ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/kick/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamKick ) + '"><i class="lp-icon lp-icon-21 lp-kick"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamLoco ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/loco/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamLoco ) + '"><i class="lp-icon lp-icon-21 lp-loco"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamMildom ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/mildom/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamMildom ) + '"><i class="lp-icon lp-icon-21 lp-mildom"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamNimo ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/nimo/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamNimo ) + '"><i class="lp-icon lp-icon-21 lp-nimotv"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamTrovo ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/trovo/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamTrovo ) + '"><i class="lp-icon lp-icon-21 lp-trovo"></i></a>' );
+		const streamKeys = {
+			Twitch: 'twitch',
+			Youtube: 'youtube',
+			Afreecatv: 'afreeca',
+			Bilibili: 'bilibili',
+			Booyah: 'booyah',
+			Cc163: 'cc',
+			Dailymotion: 'dailymotion',
+			Douyu: 'douyu',
+			Facebook: 'facebook',
+			Huomao: 'huomaotv',
+			Huya: 'huyatv',
+			Loco: 'stream',
+			Mildom: 'mildom',
+			Nimo: 'nimotv',
+			Trovo: 'trovo',
+			Kik: 'kick'
+		};
+		for ( const dataSetName in streamKeys ) {
+			const icon = streamKeys[ dataSetName ];
+			const platform = dataSetName.toLowerCase();
+			if ( timerObjectNode.dataset[ 'stream' + dataSetName ] ) {
+				streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/' + platform + '/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset[ 'stream' + dataSetName ] ) + '"><i class="lp-icon lp-icon-21 lp-' + icon + '"></i></a>' );
```

Comment:
Maybe extract the string building into a function that takes icon, dataSetName and index as parameter?